subprojects/workers/src/main/java/org/gradle/workers/WorkerExecutor.java

```diff
@@ -55,8 +56,23 @@
      * in the {@link WorkerConfiguration}.  If no idle daemons are available, a new daemon will be started.  Any errors
      * will be thrown from {@link #await()} or from the surrounding task action if {@link #await()} is not used.
      */
+    @Deprecated
     void submit(Class<? extends Runnable> actionClass, Action<? super WorkerConfiguration> configAction);
 
+    /**
+     * Submits a piece of work to be executed asynchronously.
+     *
+     * Execution of the work may begin immediately.
+     *
+     * Work configured with {@link IsolationMode#PROCESS} will execute in an idle daemon that meets the requirements set
+     * in the {@link WorkerSpec}.  If no idle daemons are available, a new daemon will be started.  Any errors
+     * will be thrown from {@link #await()} or from the surrounding task action if {@link #await()} is not used.
+     *
+     * @since 5.6
+     */
+    @Incubating
+    <T extends WorkerParameters> void execute(Class<? extends WorkerExecution<T>> workerExecutionClass, Action<? super WorkerSpec<T>> configAction);
```

Comment:
I think we should try to keep the "submit" name here.

A few ideas...
- call it `submitWork`
- call it `submitOne`
- use a builder to create a "work item" and submit that 
- (other things that we already talked about on Slack)